### ISSUE TITLE
Make duplicate slots in light DOM behave like native shadow DOM slots

### ISSUE THREAD

**Issue by nolanlawson** [salesforce/lwc#4497]
Currently, if you use duplicate `<slot>`s in the same HTML template in light DOM, e.g.:

```html
<!-- component.html -->
<template lwc:render-mode=light>
    A
    <slot></slot>
    B
    <slot></slot>
    C
</template>

```

```html
<!-- app.html -->
<template lwc:render-mode=light>
    <x-counter>
        <div>yolo</div>
    </x-counter>
</template>
```

... then it will render the slots duplicated:

```html
<x-counter>
  A
  <div>yolo</div>
  B
  <div>yolo</div>
  C
</x-counter>
```

This works regardless of [named slots](https://stackblitz.com/edit/salesforce-lwc-jxefcb?file=src%2Fmodules%2Fx%2Fcounter%2Fcounter.html,src%2Fmodules%2Fx%2Fcounter%2Fcounter.js,src%2Fmodules%2Fx%2Fapp%2Fapp.html,src%2Fmodules%2Fx%2Fapp%2Fapp.js&title=LWC%20playground) or [the default slot](https://stackblitz.com/edit/salesforce-lwc-lkmffd?file=src%2Fmodules%2Fx%2Fcounter%2Fcounter.html,src%2Fmodules%2Fx%2Fcounter%2Fcounter.js,src%2Fmodules%2Fx%2Fapp%2Fapp.html,src%2Fmodules%2Fx%2Fapp%2Fapp.js&title=LWC%20playground).

This is in [contrast](https://stackblitz.com/edit/salesforce-lwc-yxlyfw?file=src%2Fmodules%2Fx%2Fcounter%2Fcounter.html,src%2Fmodules%2Fx%2Fcounter%2Fcounter.js,src%2Fmodules%2Fx%2Fapp%2Fapp.html,src%2Fmodules%2Fx%2Fapp%2Fapp.js&title=LWC%20playground) with native shadow DOM, which renders the first slot but ignores any subsequent slots.

```html
<x-counter>
  #shadow-root
    A
    <div>yolo</div>
    B
    C
</x-counter>
```

Either way, you will get a warning during template compilation:

```
LWC1137: Invalid duplicate slot (default).
```

This is because duplicate slots are generally a component author mistake. Usually you do not want duplicates – especially in native shadow DOM, where subsequent slots are ignored.

Here is a summary of how other frameworks handle this case:

- [Lit](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy5qcyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcblxuZXhwb3J0IGNsYXNzIFNpbXBsZUdyZWV0aW5nIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIGNvbnN0cnVjdG9yKCkge1xuICAgIHN1cGVyKCk7XG4gIH1cblxuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIGh0bWxgPHNsb3Q-PC9zbG90PjxzbG90Pjwvc2xvdD5gO1xuICB9XG59XG5jdXN0b21FbGVtZW50cy5kZWZpbmUoJ3NpbXBsZS1ncmVldGluZycsIFNpbXBsZUdyZWV0aW5nKTtcbiJ9LHsibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFET0NUWVBFIGh0bWw-XG48aGVhZD5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9zaW1wbGUtZ3JlZXRpbmcuanNcIj48L3NjcmlwdD5cbjwvaGVhZD5cbjxib2R5PlxuICA8c2ltcGxlLWdyZWV0aW5nPlxuICAgIDxkaXY-eW9sbzwvZGl2PlxuICA8L3NpbXBsZS1ncmVldGluZz5cbjwvYm9keT5cbiJ9LHsibmFtZSI6InBhY2thZ2UuanNvbiIsImNvbnRlbnQiOiJ7XG4gIFwiZGVwZW5kZW5jaWVzXCI6IHtcbiAgICBcImxpdFwiOiBcIl4zLjAuMFwiLFxuICAgIFwiQGxpdC9yZWFjdGl2ZS1lbGVtZW50XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJsaXQtZWxlbWVudFwiOiBcIl40LjAuMFwiLFxuICAgIFwibGl0LWh0bWxcIjogXCJeMy4wLjBcIlxuICB9XG59IiwiaGlkZGVuIjp0cnVlfV0): subsequent slots are ignored (native shadow)
- [Svelte](https://svelte.dev/repl/1c83ef14d8f946fd9d71beacdac146aa?version=4.2.19): duplicate slots are rendered
- [Vue](https://play.vuejs.org/#eNp9UT1PwzAQ/SuWly5VMsBURZUAdYABEDB6iZJrcHF8ln0JQVX+O2dHTRupMNn3Pqzne0d551zWdyA3sgiV145EAOrcVlndOvQkHpBPC5bE3mMrVlk+I9G4UrbIJyd7eCBonSkJeBKimLVpZKDW/fYHDRZ5vCXN+UGei/ziAbmWFCq0e91kh4CWUx6jRcmKLdqAf3Gk0QYlNyIxkSuNwe+nhJHvYH3Cq0+ovq7ghzBETMlXDwF8D0rOHJW+AZro3fszDHyfyRbrzrD6H/INApouZpxk952tOfaFLqV9TLvWtvkIu4HAhtOnYtCoHJNeSd543NZfXz/Hvcluk0/Zkbe4qCx2vWwpGKQt1xiPK8Cyk/EX7z29Dw==): duplicate slots are rendered

So in other words, the current LWC light DOM behavior matches Svelte's and Vue's behavior.

**Comment by nolanlawson** [comment]
BTW this would be a breaking change and this issue is _not_ up for grabs. I am documenting the existing behavior without proposing a solution.

**Comment by nolanlawson** [comment]
Another option here is to turn the compiler warning into an error. This would make it much easier to reason about duplicate slots in either light DOM or shadow DOM, since they wouldn't exist. 🙂 

However, this would be a breaking change, and so would need to be done via API versioning.
